addons/web/static/src/js/chrome/abstract_web_client.js

```diff
@@ -389,7 +389,9 @@ var AbstractWebClient = Widget.extend(ServiceProviderMixin, {
         var data = e.data || {};
         var type = data.type || 'rainbow_man';
         if (type === 'rainbow_man') {
-            new RainbowMan(data).appendTo(this.$el);
+            if (session.show_rainbowman) {
```

Comment:
@ged-odoo _onShowEffect Implementation looks more generic and implemented such a way that later it can be improved to handle other event types, if we only support rainbowman effect then I can improve this method